internal/upgrade/upgrade.go

```diff
@@ -93,6 +98,49 @@ func Run(cfg *TestConfig) {
 			).Should(BeTrue())
 		})
 
+		It("successfully finishes control plane nodes rolling update if it is needed", func() {
+			// Check MachinesSpecUpToDate condition on KubeadmControlPlane. Repeat the check 5 times, with some waiting time,
+			// so Cluster API controllers have time to react to upgrade (it is usually instantaneous).
+			numberOfChecks := 5
+			waitBetweenChecks := 5 * time.Second
+			controlPlaneRollingUpdateStarted := false
+
+			for i := 0; i < numberOfChecks; i++ {
```

Comment:
why not using `Eventually` instead?

Reply:
Can I use `Eventually` and then if it fails just `Skip`?

Because here the rolling update might start, but it doesn't have to (e.g. in case of apps update, or some other simpler update that does not roll the control plane nodes).

So the check here checks multiple times if the rolling update has started (when it starts most probably the first check will catch it, as CAPI reacts instantly and you see the needed condition values there), and if it hasn't, then it just skips the test (assuming that rolling update is not happening).

Reply:
ah! okok. I understand. I'm not sure if there is a way to do that with Eventually.